File: src/main/java/com/vermouthx/stocker/utils/StockerNumberFormat.java
package com.vermouthx.stocker.utils;

/**
 * Shared formatters for the holdings columns. Previously duplicated across
 * StockerTableView, StockerQuoteUpdateListener and StockerManagementDialog.
 */
public final class StockerNumberFormat {

    public static final String EMPTY = "-";

    public static String formatPrice(Double price) {
        return price != null ? String.format("%.3f", price) : EMPTY;
    }

    public static Object formatHoldings(Integer holdings) {
        return holdings != null ? holdings : EMPTY;
    }

    public static Object formatNetProfit(Double currentPrice, Double costPrice, Integer holdings) {
        if (currentPrice == null || costPrice == null || holdings == null) {
            return EMPTY;
        }
        return String.format("%.3f", (currentPrice - costPrice) * holdings);
    }

    private StockerNumberFormat() {
    }
}

File: src/main/kotlin/com/vermouthx/stocker/utils/StockerNumberFormat.kt
package com.vermouthx.stocker.utils

/**
 * Shared formatters for the holdings columns. Previously duplicated across
 * StockerTableView, StockerQuoteUpdateListener and StockerManagementDialog.
 */
object StockerNumberFormat {

    const val EMPTY: String = "-"

    @JvmStatic
    fun formatPrice(price: Double?): String =
        if (price != null) String.format("%.3f", price) else EMPTY

    @JvmStatic
    fun formatHoldings(holdings: Int?): Any =
        holdings ?: EMPTY

    @JvmStatic
    fun formatNetProfit(currentPrice: Double?, costPrice: Double?, holdings: Int?): Any {
        if (currentPrice == null || costPrice == null || holdings == null) return EMPTY
        return String.format("%.3f", (currentPrice - costPrice) * holdings)
    }
}
